src/endpoint/s3/ops/s3_put_bucket_policy.js

```diff
@@ -1,30 +1,25 @@
 /* Copyright (C) 2016 NooBaa */
 'use strict';
 
-const _ = require('lodash');
 const S3Error = require('../s3_errors').S3Error;
+const { RpcError } = require('../../../rpc');
 
 /**
  * http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTpolicy.html
  */
 async function put_bucket_policy(req) {
     let policy;
     try {
-        policy = JSON.parse(JSON.stringify(req.body).toLowerCase());
+        policy = JSON.parse(JSON.stringify(req.body));
     } catch (error) {
         console.error('put_bucket_policy: Invalid JSON provided', error);
         throw new S3Error(S3Error.InvalidArgument);
     }
-    // adapting bucket policy to our schema
-    for (const statement of policy.statement) {
-        if (statement.principal) {
-            if (statement.principal.aws) statement.principal = statement.principal.aws;
-            statement.principal = _.flatten([statement.principal]);
-        }
-        if (statement.action) statement.action = _.flatten([statement.action]);
-        if (statement.resource) statement.resource = _.flatten([statement.resource]);
+    try {
+        await req.object_sdk.put_bucket_policy({ name: req.params.bucket, policy });
+    } catch (error) {
+        throw new RpcError('MALFORMED_POLICY', "Policy was not well formed or did not validate against the published schema", { detail: error.rpc_data });
```

Comment:
2 comments - 
1. Why throw MALFORMED_POLICY for every error that occurs?
2. Throwing S3 error type instead of an RPC error type is expected in these s3_.. classes